numba/core/controlflow.py

```diff
@@ -847,6 +847,7 @@ def jump(self, target, pops=0):
     def _iter_inst(self):
         for inst in self.bytecode:
             if self._use_new_block(inst):
+                self._guard_with_as(inst)
```

Comment:
The included test still passes if this line and the implementation of `_guard_with_as` are removed - is this catching a different, untested case? (I note the code here is also exactly the same as in byteflow.py).

Reply:
There's two versions of the "interpreter", https://github.com/numba/numba/commit/d36ae2af1200ab56b36e57589843974727e73196 failed https://github.com/numba/numba/runs/1821883328 as Python 3.6 goes a different route to later pythons.

Reply:
Ah, I wasn't aware - thanks for clarifying!